lvsfunc/misc.py

```diff
@@ -469,6 +472,123 @@ def colored_clips(amount: int,
     return [core.std.BlankClip(color=color, **blank_clip_args) for color in rgb_color_list]
 
 
+def save(clips: Dict[str, vs.VideoNode],
+         frames: List[int],
+         random_number: int = 0,
+         folder: bool = False,
+         dither_type: Optional[Union[str, Dither]] = None,
+         zoom: int = 1,
+         save_location: Optional[Union[str, pathlib.Path]] = None
+         ) -> None:
+    """
+    Writes frames as RGB24 PNG files for comparison websites or local comparison.
+
+    :param clips:         Dictionary of name:clip for all clips you want to save frames from. Accepts YUV or RGB clips only.
+                          Assumes all YUV-clips are limited range and all RGB-clips are full range.
+    :param frames:        List of frame numbers to save from each clip.
+    :param random_number: Number of random frames to save from each clip (Default: 0)
+    :param folder:        Whether or not to save clips in named folders instead of prefixing the files with the clip names (Default: ``False``\ )
+    :param dither_type:   ``dither_type`` override for ``vsutil.depth`` for clips >8-bit (Default: ``None``\ )
+    :param zoom:          Zoom factor for image output. Useful for comparison websites
+                          as zooming with a browser is not reliable and might not use point/nearest-neighbor (Default: 1)
+    :param save_location: An optional folder path to save the images/sub-folders to (will normally save in the same folder
+                          as the script is ran in) (Default: ``None``\ )
+    """
+    @contextlib.contextmanager
+    def _cd(newdir):
+        prevdir = os.getcwd()
+        os.chdir(os.path.expanduser(newdir))
+        try:
+            yield
+        finally:
+            os.chdir(prevdir)
+
+    if 0 in (clip.height for clip in clips.values()) or 0 in (clip.width for clip in clips.values()):
+        raise ValueError("save: variable-resolution clips not allowed")
+    if None in (clip.format for clip in clips.values()):
+        raise ValueError("save: variable-format clips not allowed")
+
+    if save_location is not None:
+        if (save_location := pathlib.Path(save_location)).exists():
+            os.chdir(save_location)
+
+    if len(clips) > 1:
+        if len(set(clip.height for clip in clips.values())) > 1 or len(set(clip.width for clip in clips.values())) > 1:
```

Comment:
`(clip.height for clip in clips.values())` (and for `width`) is computed a few lines above already and can be reused.